Add linking options for (not) banned players

stonar96 opened the pull request:
This PR adds two boolean config options to the linking.yml config file.
```yml
  # Whether or not to let players in the VANILLA banlist be able to link their accounts
  Check banned players: false
```
This can be used to even allow banned players to link their accounts.
```yml
  # Whether or not players not in the VANILLA banlist will bypass the need to link their accounts/have a sub role
  Only check banned players: false
```
This is basically the opposite of the existing option `Whitelisted players bypass check`. So instead of a whitelist you now have a blacklist (the banned players).

Example usecase: Banned users are added to a `Banned` group, which is synchronized to a `Banned` role on Discord. Only users with the `Banned` role can see the channel `#ban-appeal` but no other channels. A normal (not banned player) doesn't need to link the account. Then one can use the following setup.
```yml
Require linked account to play:
  Enabled: true
  Whitelisted players bypass check: false
  Check banned players: false
  Only check banned players: true
  Not linked message: "&cYou are banned from this server!\n\n&7To appeal your ban, you must link your &9Discord &7account.\n\n&7Send a DM to &b{BOT}&7 in the Discord server containing just &b{CODE}&7 to link your account.\n\n&7Then you can contact us in the &9#ban-appeal &7channel.\n\n&7Discord Invite ยป &b{INVITE}"
```

Vankka commented on src/main/java/github/scarsz/discordsrv/modules/requirelink/RequireLinkModule.java:
```diff
@@ -67,13 +67,27 @@ private void check(String eventType, EventPriority priority, String playerName,
+                    banned = true;
+                }
+                if (onlyCheckBannedPlayers && !banned) {
+                    DiscordSRV.debug("Player " + playerName + " is bypassing link requirement, player is not banned");
```
Should say something like `(beacuse Only check banned players is enabled)`

stonar96 replied:
Done.

Vankka reviewed (commented):
Comments for other languages (except Estonian) can be translated using Google Translate, we also have a bot that can do this on our Discord (https://snapshot.discordsrv.com), just get the attention of one of us and ask for the ability to use it

stonar96 commented:
Changed base branch from master to develop. (Sorry for not reading carefully.)
Rebased (without changes).
Improved debug messages.
Added translations.

Vankka commented on src/main/java/github/scarsz/discordsrv/modules/requirelink/RequireLinkModule.java:
```diff
@@ -192,10 +206,15 @@ public void noticePlayerUnlink(Player player) {
             boolean whitelisted = Bukkit.getServer().getWhitelistedPlayers().stream().map(OfflinePlayer::getUniqueId).anyMatch(u -> u.equals(player.getUniqueId()));
             if (whitelisted) {
-                DiscordSRV.debug("Player " + player.getName() + " is bypassing link requirement, player is whitelisted");
+                DiscordSRV.debug("Player " + player.getName() + " is bypassing link requirement beacuse Only check banned players is enabled");
```
Changed the debug message in the wrong spot? Looks like this was suppose to be on line 215 instead

stonar96 replied:
Oops. You are right. Thanks!
